src/routers/alpha-router/alpha-router.ts

```diff
@@ -1995,17 +2019,33 @@ export class AlphaRouter
         )
       : Promise.resolve(null);
 
-    const [usdPool, nativeQuoteTokenV3Pool, nativeAmountTokenV3Pool] =
-      await Promise.all([
-        usdPoolPromise,
-        nativeQuoteTokenV3PoolPromise,
-        nativeAmountTokenV3PoolPromise,
-      ]);
+    const nativeGasTokenV3PoolPromise =
+      providerConfig?.gasToken &&
+      !providerConfig?.gasToken.equals(nativeCurrency)
+        ? getHighestLiquidityV3NativePool(
+            providerConfig?.gasToken,
+            this.v3PoolProvider,
+            providerConfig
+          )
+        : Promise.resolve(null);
+
+    const [
+      usdPool,
+      nativeQuoteTokenV3Pool,
+      nativeAmountTokenV3Pool,
+      nativeGasTokenV3Pool,
+    ] = await Promise.all([
+      usdPoolPromise,
+      nativeQuoteTokenV3PoolPromise,
+      nativeAmountTokenV3PoolPromise,
+      nativeGasTokenV3PoolPromise,
+    ]);
 
     const pools: LiquidityCalculationPools = {
       usdPool: usdPool,
       nativeQuoteTokenV3Pool: nativeQuoteTokenV3Pool,
       nativeAmountTokenV3Pool: nativeAmountTokenV3Pool,
+      nativeGasTokenV3Pool: nativeGasTokenV3Pool,
```

Comment:
nit: I think the readability will benefit from adding some comments, explaining this new pool, although it's called native, is not to be confused with the native ETH as gas token.

Reply:
Yep agree, actually changing the naming throughout to be `nativeAnd<token>` to hopefully make it more explicit that it refers to a single V3 pool between the native currency and some other token